pallets/gas/src/tests.rs

```diff
@@ -629,3 +669,83 @@ fn gas_free_after_consumed() {
         assert_eq!(Gas::get_limit(root_msg_id), Ok(None));
     })
 }
+
+#[test]
+fn test_imbalances_drop() {
+    new_test_ext().execute_with(|| {
+        let pos_imb = PositiveImbalance::<Balance, TotalIssuanceWrap<Test>>::new(100);
+        assert_eq!(TotalIssuance::<Test>::get(), None);
+        drop(pos_imb);
+        assert_eq!(TotalIssuance::<Test>::get(), Some(100));
+        let neg_imb = NegativeImbalance::<Balance, TotalIssuanceWrap<Test>>::new(50);
+        assert_eq!(TotalIssuance::<Test>::get(), Some(100));
+        let new_neg = NegativeImbalance::<Balance, TotalIssuanceWrap<Test>>::new(30).merge(neg_imb);
+        assert_eq!(TotalIssuance::<Test>::get(), Some(100));
+        drop(new_neg);
+        assert_eq!(TotalIssuance::<Test>::get(), Some(20));
+    })
+}
+
+#[test]
+fn catch_value_all_blocked() {
+    new_test_ext().execute_with(|| {
+        // All nodes are blocked
+        let root = random_node_id();
+        let spec_1 = random_node_id();
+        let spec_2 = random_node_id();
+        let spec_3 = random_node_id();
+
+        let res = Gas::create(ALICE, root, 10000);
+        assert!(res.is_ok());
+        drop(res);
+        assert_eq!(Gas::total_supply(), 10000);
+        assert_ok!(Gas::split(root, random_node_id()));
+        assert_ok!(Gas::split(root, random_node_id()));
+
+        assert_ok!(Gas::split_with_value(root, spec_1, 100));
+        assert_ok!(Gas::split(spec_1, random_node_id()));
+        assert_ok!(Gas::split(spec_1, random_node_id()));
+
+        assert_ok!(Gas::split_with_value(root, spec_2, 100));
+        assert_ok!(Gas::split(spec_2, random_node_id()));
+        assert_ok!(Gas::split(spec_2, random_node_id()));
+
+        assert_ok!(Gas::split_with_value(root, spec_3, 100));
+        assert_ok!(Gas::split(spec_3, random_node_id()));
+        assert_ok!(Gas::split(spec_3, random_node_id()));
+
+        // None of ops will catch the value
+        assert!(matches!(Gas::consume(root).unwrap(), None));
+        assert!(matches!(Gas::consume(spec_1).unwrap(), None));
+        assert!(matches!(Gas::consume(spec_2).unwrap(), None));
+        assert!(matches!(Gas::consume(spec_3).unwrap(), None));
+
+        assert_eq!(Gas::total_supply(), 10000);
+    })
+}
+
+#[test]
+fn catch_value_all_catch() {
+    new_test_ext().execute_with(|| {
+        // All nodes are blocked
+        let root = random_node_id();
+        let spec_1 = random_node_id();
+        let spec_2 = random_node_id();
+        let spec_3 = random_node_id();
+
+        let res = Gas::create(ALICE, root, 10000);
+        assert!(res.is_ok());
+        drop(res);
```

Comment:
```suggestion
        assert_ok!(Gas::create(ALICE, root, 10000));
```

Reply:
dunno why, when assert_ok! is used, value returned after expression in it is not dropped. Will use the `assert!(Gas::create(ALICE, root, 10000).is_ok());` instead